### PR TITLE
Fix(Stats): Stats now work offline and fixed multiple bugs.

### LINKED ISSUE #236: Fix the stats update offline.
**Description:**

Currently there are some problems with the stats, they are not updated correctly when switching from online to offline, this needs to be addressed, along with some incoherences in other parts that involve them.

**Criteria:**

- [x] The stats fully handle the offline/online switch with persistence.
- [x] The study session screen handles a big number of subjects.
- [x] The plot in the stats screen is correctly aligned with the days, and plot the current day in evidence.
- [x] Study time updated in the study session screen when doing offline/online with persistence.

### PR DESCRIPTION
**Pull request by Adessrrrr**
## Summary
Stats and study session screens were inconsistent and had UX issues:

“Total study time”/stats displayed different values across Home, Study, and Stats screens after going offline.
Subject chips overflowed and pushed the Pomodoro timer off-screen.
Weekly charts used confusing D-6 … Today labels and a “Cloud” scenario button that did nothing.
The “Time per subject” section did not clearly indicate it was week-scoped.
The stats were not correctly updated when switching between offline and online.

Now every one of these problems is fixed.

## What’s in this PR

- Make StudySession scrollable + wrap subject chips in a FlowRow.
- Replace D-6…Today labels with real weekdays (Mon–Sun).
- Remove the useless “Cloud” scenario button.
- Add hint “Only includes study time for the current week.”
- Use UserStatsRepository as the source for total study time across screens.
- Keep StudyStats for weekly breakdown (per subject, 7-day progression).

## Implementation Notes

- No Firestore schema changes.
- Weekly stats update as before; total study time now comes from UserStatsRepository for consistency.
- UI changes only: scroll support, weekday labels, legend hint, scenario cleanup.
- Firestore writes remain merged; no added transactions.

## Testing
**Unit Tests (25 tests):**

- StatsViewModelTest (12 tests): State management, scenario selection, repository interactions, index clamping.
- StudyStatsModelTest (13 tests): Data class integrity, copy functionality, edge cases (empty/zero/large values).

**UI/Instrumented Tests (34 tests):**

- StatsScreenTest (18 tests): Component rendering, time formatting, charts, user interactions.
- StatsRouteTest (4 tests): Route composable, ViewModel integration, scenario selection.
- StatsChartComponentsTest (9 tests): Pie chart, bar chart, legend, edge cases (empty data, zero values).

## Screenshots / Demos
image
image
image

## Checklist
- [x] The stats fully handle the offline/online switch with persistence.
- [x] The study session screen handles a big number of subjects.
- [x] The plot in the stats screen is correctly aligned with the days, and plot the current day in evidence.
- [x] Study time updated in the study session screen when doing offline/online with persistence.

## Notes
The help of an A.I assistant has been used for parts of this PR and parts of the code in this PR.

## Issue Reference
Closes #236

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/session/StudySessionScreen.kt [hunk 1]
```diff
@@ -194,10 +212,13 @@ private fun SubjectsSection(
     Spacer(Modifier.height(SUBJECT_SECTION_SPACING_DP.dp))
 
     if (uiState.subjects.isNotEmpty()) {
-      Row(horizontalArrangement = Arrangement.spacedBy(SUBJECT_CHIP_SPACING_DP.dp)) {
+      FlowRow(
+          horizontalArrangement = Arrangement.spacedBy(SUBJECT_CHIP_SPACING_DP.dp),
+          verticalArrangement = Arrangement.spacedBy(SUBJECT_CHIP_SPACING_DP.dp),
+      ) {
```
Comment (alaaae) [inline app/src/main/java/com/android/sample/ui/session/StudySessionScreen.kt:218]: Very good for not using magic number !

File: app/src/main/java/com/android/sample/ui/stats/StatsScreen.kt [hunk 2]
```diff
@@ -122,7 +145,17 @@ fun StatsScreen(
...
                     fontWeight = FontWeight.SemiBold,
                     color = onSurf)
+
+                Spacer(Modifier.height(8.dp))
+
+                Text(
+                    text = stringResource(R.string.stats_subjects_this_week_hint),
+                    style = MaterialTheme.typography.bodySmall,
+                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.8f),
+                )
```
Comment (alaaae) [inline app/src/main/java/com/android/sample/ui/stats/StatsScreen.kt:156]: Magic number ! Please change it

**Review by alaaae — CHANGES_REQUESTED** [review]
Very clean code overall. 
Amazing coverage with no new issues, good job ! 
No hardcoded strings or colors, well done !
Just some very few magic numbers. Change them and you will be ready to merge.

**Review by alaaae — APPROVED** [review]
Changes requests done ! Thank you !
